pkg/skaffold/runner/deployer.go

```diff
@@ -66,11 +66,45 @@ func (d *deployerCtx) JSONParseConfig() v1.JSONParseConfig {
 
 // GetDeployer creates a deployer from a given RunContext and deploy pipeline definitions.
 func GetDeployer(ctx context.Context, runCtx *runcontext.RunContext, labeller *label.DefaultLabeller) (deploy.Deployer, error) {
+	deployerCfg := runCtx.Deployers()
+
 	if runCtx.Opts.Apply {
+		helmNamespaceValue := ""
+		helmReleasesWithDiffNamespacesFound := false
+		nonHelmDeployFound := false
```

Comment:
Can we simply the if conditions and flows by using a map for seen namespaces and check if their length is 1?

Reply:
Definitely, I've made the suggested change